cypress/unit/store-mutations-setDataDictionary.cy.js

```diff
@@ -0,0 +1,75 @@
+import { mutations } from "~/store/index-refactor";
+
+let state = {};
+
+describe("setDataDictionary", () => {
+
+    beforeEach(() => {
+
+        // Setup
+        state = {
+
+            dataDictionary: {
+
+                "age": {
+
+                    "Description": "age of the participant",
+                    "Units": "years"
+                },
+                "sex": {
+
+                    "Description": "sex of the participant as reported by the participant",
+                    "Levels": {
+                        "M": "male",
+                        "F": "female"
+                    }
+                }
+            },
+
+            dataTable: [
+
+                {
+                    "age": "value1", "sex": "value2"
+                },
+                {
+                    "age": "value3", "sex": "value4"
+                }
+            ]
+        };
+    });
+
+    it("New columns in the uploaded dictionary should not be added to the state data dictionary?", () => {
+
+        // Act
+        mutations.setDataDictionary(state,
+            { "group": { "key1": "value1" } },
+            ["age", "sex"]);
+
+        // Assert
+        expect(state.dataDictionary).to.not.contain.keys("group");
+    });
+
+    it("New string/string key/value pairs for columns should replace values in the state data dictionary", () => {
+
+        // Act
+        mutations.setDataDictionary(state,
+            { "age": { "Units": "minutes" } },
+            ["age", "sex"]);
+
+        // Assert
+        expect(state.dataDictionary["age"]["Units"]).to.equal("minutes");
+    });
+
+    it("New string/object key/value pairs in the uploaded dictionary should update/retain values in the state data dictionary", () => {
+
+        // Act
+        mutations.setDataDictionary(state,
+            { "sex": { "Levels": { "M": "male", "F": "female", "NB": "non-binary" }} },
+            ["age", "sex"]);
+
+        // Assert
+        expect(state.dataDictionary["sex"]["Levels"]).to.contain.keys("M");
```

Comment:
why not just assert `deep.equal` of the entire object we expect in `"sex"`?

Reply:
Potato/potata, basically. But if we wanted to change the test input then a `deep.equal` with a locally defined var for the new input makes sense. I'll make the change.

Reply:
Yeah, I think it makes more sense if the thing you're passing to the mutation is different in some places from the original dataDictionary. Then we can see whether the merged dataDictionary really updated the different fields and kept the ones that were not specified.